00-dev-environment/refresh_access_token.py

```diff
@@ -0,0 +1,57 @@
+import boto3
+import sys
+import configparser
+from pathlib import Path
+from botocore.exceptions import ClientError
+
+"""
+This script takes a profile and authenticator code to get a session token automatically.
+Usage "python3 refresh_access_token.py MFA_TOKEN_VALUE"
+Usage if you have multiple mfa accounts (unimplemented) "python3 refresh_access_token.py USER_IAM_NAME USER_AWS_ACCOUNT INPUT_PROFILE_NAME MFA_TOKEN_VALUE"
+"""
+
+# if you have more than 1 mfa account, these will need to be params as well. maybe store in a local config file for easier re-use!?
+USER_IAM_NAME = "josh.dix.labs"
+USER_AWS_ACCOUNT = "324320755747"
+USER_MFA_DEVICE_ARN = "arn:aws:iam::" + USER_AWS_ACCOUNT + ":mfa/" + USER_IAM_NAME
+AWS_CREDENTIAL_FILE_LOCATION = str(Path.home()) + "/.aws/credentials"
+
+AWS_CRED_FILE_ACCESS_KEY_STRING = "aws_access_key_id"
+AWS_CRED_FILE_SECRET_KEY_STRING = "aws_secret_access_key"
+AWS_CRED_FILE_SESSION_TOKEN_STRING = "aws_session_token"
+
+INPUT_PROFILE_NAME = "stelligent_labs"
+SESSION_PROFILE_SUFFIX = "_temp"
+SESSION_PROFILE_NAME = INPUT_PROFILE_NAME + SESSION_PROFILE_SUFFIX
+INPUT_MFA_CODE = sys.argv[1]
+
+def program_failure(failure_message):
+    print(failure_message)
+    sys.exit(1)
+
+def main():
+    aws_cred_conf_parser = configparser.ConfigParser()
+    aws_cred_conf_parser.read(AWS_CREDENTIAL_FILE_LOCATION)
+
+    if INPUT_PROFILE_NAME in aws_cred_conf_parser.sections():
+        try:
+            sts_client = boto3.client("sts", aws_access_key_id=aws_cred_conf_parser[INPUT_PROFILE_NAME]['aws_access_key_id'], aws_secret_access_key=aws_cred_conf_parser[INPUT_PROFILE_NAME]['aws_secret_access_key'])
```

Comment:
Elsewhere you (correctly) use AWS_CRED_FILE for the conf key; could be done here as well I believe. You do not need to spend any time changing this.

Reply:
I think this is correct unless I'm not understanding. The difference is the client is created with existing 'static' creds based on the "input" (though I left this staticly configured at the end of the day), then I store 'temp' creds named with the suffix 'temp'.

tl;dr: I use the static creds to create the temp creds. Static is 'input',  written is 'session'/'temp'. This is largely set up to handle multiple input profiles.